src/action_runner/action_runner.cpp

```diff
@@ -192,7 +189,23 @@ int WINAPI WinMain(HINSTANCE, HINSTANCE, LPSTR, int)
         {
             return 0;
         }
-        return !install_dotnet();
+
+        using installer_link_and_filename_t = std::pair<std::wstring_view, std::wstring_view>;
+        std::array<installer_link_and_filename_t, 2> dotnet_installers = {
+            installer_link_and_filename_t{ L"https://download.visualstudio.microsoft.com/download/pr/fa69f1ae-255d-453c-b4ff-28d832525037/51694be04e411600c2e3361f6c81400d/dotnet-runtime-3.0.3-win-x64.exe",
+                                           L"dotnet-runtime-3.0.3-win-x64.exe" },
+            installer_link_and_filename_t{ L"https://download.visualstudio.microsoft.com/download/pr/c525a2bb-6e98-4e6e-849e-45241d0db71c/d21612f02b9cae52fa50eb54de905986/windowsdesktop-runtime-3.0.3-win-x64.exe",
+                                           L"windowsdesktop-runtime-3.0.3-win-x64.exe" }
+        };
+
+        for (const auto [installer_link, installer_filename] : dotnet_installers)
+        {
+            if (!install_dotnet(installer_link, installer_filename))
```

Comment:
`install_dotnet` returns when the installation is completed?

Reply:
yes